class-wp-options-importer.php

```diff
@@ -0,0 +1,857 @@
+<?php
+/**
+ * The main class file.
+ *
+ * @package Options_Importer
+ */
+
+/**
+ * The main class.
+ */
+class WP_Options_Importer {
+
+	/**
+	 * Stores the singleton instance.
+	 *
+	 * @access private
+	 *
+	 * @var object
+	 */
+	private static $instance;
+
+	/**
+	 * The attachment ID.
+	 *
+	 * @access private
+	 *
+	 * @var int
+	 */
+	private $file_id;
+
+	/**
+	 * The transient key template used to store the options after upload.
+	 *
+	 * @access private
+	 *
+	 * @var string
+	 */
+	private $transient_key = 'options-import-%d';
+
+	/**
+	 * The plugin version.
+	 */
+	const VERSION = 7;
+
+	/**
+	 * The minimum file version the importer will allow.
+	 *
+	 * @access private
+	 *
+	 * @var int
+	 */
+	private $min_version = 2;
+
+	/**
+	 * Stores the import data from the uploaded file.
+	 *
+	 * @access public
+	 *
+	 * @var array
+	 */
+	public $import_data;
+
+	/**
+	 * Don't do anything, needs to be initialized via instance() method.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Die if attempting to clone this class.
+	 */
+	public function __clone() {
+		wp_die( "Please don't __clone WP_Options_Importer" );
+	}
+
+	/**
+	 * Die if attempting to wakup this class.
+	 */
+	public function __wakeup() {
+		wp_die( "Please don't __wakeup WP_Options_Importer" );
+	}
+
+	/**
+	 * Creates and returns the singleton instance.
+	 *
+	 * @return \WP_Options_Importer The singleton instance.
+	 */
+	public static function instance() {
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new WP_Options_Importer();
+			self::$instance->setup();
```

Comment:
The `$object->setup()`, at least, should also be called after `plugins_loaded`, so that other plugins/theme code has a chance to interact on earlier hook priority if needed.